website/source/guides/tls.html.md

```diff
@@ -0,0 +1,425 @@
+---
+layout: "guides"
+page_title: "Securing Nomad with TLS"
+sidebar_current: "guides-tls"
+description: |-
+  Securing Nomad's cluster communication with TLS is important for both
+  security and easing operations. Nomad can use mutual TLS (mTLS) for
+  authenticating for all HTTP and RPC communication.
+---
+
+# Securing Nomad with TLS
+
+Securing Nomad's cluster communication is not only important for security but
+can even ease operations by preventing mistakes and misconfigurations. Nomad
+optionally uses mutual [TLS][tls] (mTLS) for all HTTP and RPC communication.
+Nomad's use of mTLS provides the following properties:
+
+* Prevent unauthorized Nomad access
+* Prevent observing or tampering with Nomad communication
+* Prevent client/server role or region misconfigurations
+* Prevent other services from masquerading as Nomad agents
+
+Preventing region misconfigurations is a property of Nomad's mTLS not commonly
+found in the TLS implementations on the public Internet.  While most uses of
+TLS verify the identity of the server you are connecting to based on a domain
+name such as `example.com`, Nomad verifies the node you are connecting to is in
+the expected region and configured for the expected role (e.g.
+`client.us-west.nomad`). This also prevents other services who may have access
+to certificates signed by the same private CA from masquerading as Nomad
+agents. If certificates were identified based on hostname/IP then any other
+service on a host could masquerade as a Nomad agent.
+
+Correctly configuring TLS can be a complex process, especially given the wide
+range of deployment methodologies. If you use the sample
+[Vagrantfile][vagrantfile] from the [Getting Started Guide][guide-install] - or
+have [cfssl][cfssl] and Nomad installed - this guide will provide you with a
+production ready TLS configuration.
+
+~> Note that while Nomad's TLS configuration will be production ready, key
+   management and rotation is a complex subject not covered by this guide.
+   [Vault][vault] is the suggested solution for key generation and management.
+
+## Creating Certificates
+
+The first step to configuring TLS for Nomad is generating certificates. In
+order to prevent unauthorized cluster access, Nomad requires all certificates
+be signed by the same Certificate Authority (CA). This should be a _private_ CA
+and not a public one like [Let's Encrypt][letsencrypt] as any certificate
+signed by this CA will be allowed to communicate with the cluster.
+
+~> Nomad certificates may be signed by different intermediate CAs as long as
```

Comment:
This sentence seems confused. Rather than loading the intermediate certs along the chain into the `ca_file` of all hosts, the intermediate certs should be appended to the `cert_file`, as noted in https://golang.org/pkg/crypto/tls/#LoadX509KeyPair.